doc/man3/SSL_read_early_data.pod

```diff
@@ -168,6 +168,30 @@ In the event that the current maximum early data setting for the server is
 different to that originally specified in a session that a client is resuming
 with then the lower of the two values will apply.
 
+=head1 NOTES
+
+The whole purpose of early data is to enable a client to start sending data to
+the server before a full round trip of network traffic has occurred. Application
+developers should ensure they consider optimisation of the underlying TCP socket
+to obtain a performant solution. For example Nagle's algorithm is commonly used
+by operating systems in an attempt to avoid lots of small TCP packets. In many
+scenarios this is beneficial for performance, but it does not work well with the
+early data solution as implemented in OpenSSL. In Nagle's algorithm the OS will
+buffer outgoing TCP data if a TCP packet has already been sent which we have not
+yet received an ACK for from the peer. The buffered data will only be
+transmitted if enough data to fill an entire TCP packet is accumulated, or if
+the ACK is received from the peer. The initial ClientHello will be sent as the
+first TCP packet, causing the early application data from calls to
+SSL_write_early_data() to be buffered by the OS and not sent until an ACK is
+received for the ClientHello packet. This means the early data is not actually
```

Comment:
Personally I think the previous sentence makes it clear. But if you prefer I could add "(unless enough data is written to fill an entire TCP packet)".